### PR TITLE
helper component for typography plugin

### PR DESCRIPTION
**Pull request by ssahara**
Hi @LarsGit223 

This PR is to show another way to implement odt export feature to typography plugin. I appreciate it if you could test the combination of [odt2 branch](https://github.com/ssahara/dw-plugin-typography/tree/odt2) (typography plugin) and this PR (typography branch of odt plugin).

I think this way will also applicable other plugins that support odt export feature, and make ease your development to increase ODT supportive plugins.

Satoshi

### CODE REVIEW

**Comment by LarsGit223** [comment]
Hi Satoshi,

the PR seems to work so far.

Moving plugin specific code to a helper class in the ODT plguin is a general decision which would also affect other plugins. I decided to write an e-mail to the mailinglist to have a vote/decision if we like to go that way or not. But to me it seems that there would be some advantages if we move the code to helper plugins in the ODT plugin.

We will have to wait what the outcome of the diiscussion is. Of course you can add your own opinion/ideas to my mail.

So, right now I will not merge this but the PR seems to work. Thanks.

Cheers,
Lars

**Comment by ssahara** [comment]
Hello @LarsGit223 
First of all, please understand that I do not have any idea to oppose ODT support. I just feel it seems difficult to keep odt_render work well with possible plugin code updates, because I do not have enough knowledge of ODT file format. 

At least I would like to separate odt specific code to other helper file for better maintainability. 
The separated helper can be stored in
1) each plugin helper folders (distributed model, `class helper_plugin_typography_odt`), or 
2) ODT plugin helper folder (central model, `class helper_plugin_odt_typography`).

The helper code will be identical except class name. Method 1 is enough for me, however it might be possible to prepare common helper for some odt supportive plugins and bring something advantage for development of ODT plugin.

Thanks.
Satoshi

**Comment by LarsGit223** [comment]
Hi Satoshi,
after seeing the answer from splitbrain to my mail, I would prefer option 1 for the reason to share work amongst people.

> I just feel it seems difficult to keep odt_render work well with possible plugin code updates, because I do not have enough knowledge of ODT file format. 

Well, I know that's not easy. But I try to not change the ODT code in that way, that old functions do not work anymore. Or did you have trouble with the old typography code and latest ODT plugin code?

Anyway, I still need to write a lot of new functions because the ODT plugin is not on that level that I want to achive. But, in general I will keep the method of using CSS properties as the input for ODT functions. But sometimes as I go working on the ODT plugin I find problems that I can't change without changing the interface. So, currently there is unfortunately quite a lot of movement in the ODT plugin.

Also, I did not write that much documentation right now, because I do not want to re-write it in case there are bigger changes some time in the future. On the other hand I commit my code regularly as I want  to share the code.

So, I'm sorry if that's hard to follow. If you got questions please let me know. I'll be glad to help you.

Cheers,
Lars

**Comment by ssahara** [comment]
Hello @LarsGit223 
OK. collaborative development is fun exactly! I withdraw this PR.
Anyway I prefer option 1, I will move odt code to separated helper components when I have a time to de so. I will inform you when I made some change to typography plugin.

Good luck for ODT plugin development!
Satoshi

**Comment by LarsGit223** [comment]
Hi @ssahara,
OK, thanks. Working with you was fun like always.

Cheers,
Lars

**Comment by ssahara** [comment]
Hi @LarsGit223
New Typography plugin 2015-06-05 released.
